### ISSUE TITLE
0iF controller

### ISSUE THREAD

**Issue by cilpatel** [TrakHound/Fanuc-MTConnect-Agent#6]
Can I connect to a Fanuc 0iF controller using this agent? If yes, then please guide me how to connect?

**Comment by intolerance** [comment]
Does the control have a FOCAS or FOCAS2 option?  If so, then there is a good possibility that it does.

[Setup FOCAS/FOCAS2 Google Results](https://www.google.com/search?q=how+to+configure+fanuc+focas)

**Comment by cilpatel** [comment]
Yes it has FOCAS2. I have already setup it on the controller. I need to know how to setup the agent in my pc to connect to the controller via FOCAS2. Thanks in advance.

**Comment by intolerance** [comment]
Firstly, have you configured your machine using the `Fanuc MTConnect Configurator` to enable the Software Adapter for your machine? If not, you can access the `Fanuc MTConnect Agent Configurator` in the `C:\Program Files (x86)\Fanuc MTConnect Agent\Configurator` directory (if you don't have a shortcut).

Once you have the configurator open, you may want to select the 30i family for this controller since it has FOCAS2. Enter the IP address, manufacturer, model, serial and description for your machine and click Add. The Software adapter service will start automatically. 

Next, you need to try and hit the agent that comes with the Fanuc MTConnect Agent. Try visiting http://localhost:5000/current in your web browser. If you are seeing your machine and it's current data then everything is working. 

If that does not work, 

- Try opening up the `agent.cfg` in `C:\Program Files (x86)\Fanuc MTConnect Agent\Agent\bin` and check the `port` and try again using http://localhost:port.
- Ensure that the `agent` service is running by going opening `services.msc` and look for `Fanuc MTConnect Agent` to ensure that it is running then try again.

**Comment by cilpatel** [comment]
The Fanuc MTConnect Agent service and the adapter service are running on my PC. On opening http://localhost:5000/current in my browser, it shows the device I added in the format same as the template devices.xml file but all the values are also same as the template file.
The port number in the agent.cfg file is 7880 and I have set the same in my machine FOCAS ethernet settings too. But there is no response from the browser on opening http://localhost:7880

**Comment by rubenruru** [comment]
Hi, I am having the same Issue,
I get the same result, I have set the static IP and the subnet mask correctly.
Did you solve the issue? what did you do to solve it?

**Comment by Castolus** [comment]
> Both the Fanuc MTConnect Agent service as well as the adapter service are running on my PC.
> On opening http://localhost:5000/current in my browser, it shows the device I added in the format same as the template devices.xml file but all the values are also same as the template file.
> 
> The agent port number in the agent.cfg file is 5000 and adapter port number is 7879.
> 
> I have set my FOCAS2 IP = 192.168.0.100 and Port Number = 8193 on my controller.
> 
> On debugging the adapter, I found that the adapter gives the result=-16 while trying to connect to the machine at 192.168.0.100 port 8193.
> 
> PFA - Command Prompt snippet for the same.
> 
> ![tempsnip](https://user-images.githubusercontent.com/42149703/45252075-bd63aa00-b36e-11e8-9033-f39224b1ce5a.png)

Hi,
Maybe I´m wrong but, shouldn´t your FOCAS IP be 192.168.**1**.100?? Otherwise it won´t connect since you wrote 192.168.**0**.100.

**Comment by Castolus** [comment]
> Both the Fanuc MTConnect Agent service as well as the adapter service are running on my PC.
> On opening http://localhost:5000/current in my browser, it shows the device I added in the format same as the template devices.xml file but all the values are also same as the template file.
> 
> The agent port number in the agent.cfg file is 5000 and adapter port number is 7879.
> 
> I have set my FOCAS2 IP = 192.168.0.100 and Port Number = 8193 on my controller.
> 
> On debugging the adapter, I found that the adapter gives the result=-16 while trying to connect to the machine at 192.168.0.100 port 8193.
> 
> PFA - Command Prompt snippet for the same.
> 
> ![tempsnip](https://user-images.githubusercontent.com/42149703/45252075-bd63aa00-b36e-11e8-9033-f39224b1ce5a.png)

Hi,

just found this. Maybe you find it helpfull:

![image](https://user-images.githubusercontent.com/74269149/127439719-d163739b-3eb6-43c0-aa78-32904fa143a5.png)
"Result: -16" means wiring error. Check connections.
